src/org/jitsi/impl/neomedia/rtp/RTPEncodingImpl.java

```diff
@@ -279,11 +279,15 @@ public boolean matches(byte[] buf, int off, int len)
             return false;
         }
 
+        if (temporalId == -1)
+        {
+            return true;
+        }
+
         int tid = track.getMediaStreamTrackReceiver()
             .getStream().getTemporalID(buf, off, len);
 
-        return (tid == -1 && ArrayUtils.isNullOrEmpty(dependencyEncodings))
-                || tid == temporalId;
+        return tid == temporalId;
```

Comment:
No, the array check was actually handling the case you described before. In practice, I think we should be fine with this code.